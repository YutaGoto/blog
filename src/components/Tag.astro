---
import Link from "@/components/global/Link.astro"

interface Props {
  tag: string
  className?: string
  withLink?: boolean
  active?: boolean
}

const { tag, className, withLink = true, active = false } = Astro.props
---

<span
  class={`${className} rounded ${
    active ? "bg-emerald-600" : "bg-emerald-800"
  } py-1 px-2 text-sm text-gray-100`}
>
  {
    withLink ? (
      <Link href={`/tags/${tag}`} astro-data-history="push">
        {tag}
      </Link>
    ) : (
      tag
    )
  }
</span>
