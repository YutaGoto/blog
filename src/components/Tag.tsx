import Link from "next/link"
import React from "react"

interface TagProps {
  tag: GatsbyTypes.Maybe<string>
  className?: string
  withLink?: boolean
}

export const Tag = ({ tag, className = "", withLink = true }: TagProps) => {
  return (
    <span
      className={`${className} rounded bg-emerald-800 py-1 px-2 text-sm text-gray-100`}
    >
      {withLink ? <Link href={`/tags/${tag}`}>{tag}</Link> : tag}
    </span>
  )
}
