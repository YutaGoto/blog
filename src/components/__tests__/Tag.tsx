import React from "react"
import { create } from "react-test-renderer"
import { render } from "@testing-library/react"
import { Tag } from "../Tag"

describe("Tag", () => {
  const tag = "タグ"

  it("renders correctly", () => {
    const tree = create(<Tag tag={tag} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("display the span element", () => {
    const { container } = render(<Tag tag={tag} />)
    expect(container.querySelector(`span`)).toBeInTheDocument()
  })
})
