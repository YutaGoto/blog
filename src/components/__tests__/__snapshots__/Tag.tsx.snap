// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders correctly 1`] = `
<span
  className=" py-1 px-2 rounded text-sm bg-emerald-800"
>
  <a
    href="/tags/タグ"
  >
    タグ
  </a>
</span>
`;
